src/components/KTable/KTable.cy.ts

```diff
@@ -149,6 +149,198 @@ describe('KTable', () => {
     })
   })
 
+  describe('data changes as expected', () => {
+    it('when clicking a specific page number for non-offset pagination', () => {
+      mount(KTable, {
+        propsData: {
+          testMode: 'true',
+          initialFetcherParams: {
+            page: 1,
+            pageSize: 1,
+          },
+          fetcher: () => {
+            return {
+              data: options.data,
+              total: options.data.length,
+            }
+          },
+          isLoading: false,
+          headers: options.headers,
+          paginationPageSizes: [1],
+          hidePaginationWhenOptional: false,
+        },
+      })
+
+      cy.getTestId('k-table-pagination').should('be.visible')
+      cy.getTestId('k-select-input').click()
+      cy.getTestId('k-select-item-1').click({ multiple: true, force: true })
+      cy.getTestId('next-btn').find('a').click()
+      cy.get('.pagination-button.active').should('contain.text', 2 + '')
+      cy.get('.k-table').find('tr').should('have.length', 4)
+    })
+
+    it('when clicking arrows for offset based pagination', () => {
+      const offsetPaginationHeaders = [
```

Comment:
suggestion: Can you move you mock data logic into a separate file in the mocks folder? https://github.com/Kong/kongponents/tree/main/mocks

Reply:
Done